### DISCUSSION TITLE
Widget "checkupdates"

### DISCUSSION THREAD

**Discussion by Breizil29** [qtile/qtile#4517] [Q&A]
Hello everyone,

I used this checkupdates widget:

widget.CheckUpdates(
font="Noto Sans",
fontsize=15,
update_interval =3600,
distro ='Arch',
custom_command='checkupdates;yay',
display_format='  {updates} ',
foreground=colors[0],
background=colors[20],
execute = 'alacritty -e yay',
),

And it shows me 7 updates and yet my system is up to date.

test with theses  2 commands : 

custom_command='checkupdates;yay -Qum',
custom_command='checkupdates;yay -Qua',

and the widget does not appear

**Comment by elParaguayo** [comment]
The widget works by taking the command and counting the number of lines output by that command.  

Also, see notes from the docs:
```
It is common for package managers to return a non-zero code when there are no
updates. As a result, the widget will treat *any* error as if there are no updates.
If you are using a custom commmand/script, you should therefore ensure that it
returns zero when it completes if you wish to see the output of your command.

In addition, as no errors are recorded to the log, if the widget is showing no
updates and you believe that to be incorrect, you should run the appropriate
command in a terminal to view any error messages.
```
